src/components/cryptoSuggestCard/CryptoSuggestCard.tsx

```diff
@@ -0,0 +1,104 @@
+import styles from "./CryptoSuggestCard.module.css";
+import axios from "axios";
+
+import {
+  CryptoItem,
+  CryptoCardProps,
+  CryptoItemWithPrice,
+} from "../../data/interfaces";
+import PriceModal from "../priceModal/PriceModal";
+import { useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+
+function CryptoSuggestCard({
+  cryptoItem,
+  cryptoListToSuggest,
+  cryptoListPinned,
+  modalVisibility,
+  setCryptoListToSuggest,
+  setCryptoListPinned,
+  setModalVisibility,
+  wrapperPrimaryRef,
+}: CryptoCardProps) {
+  // STATES
+  const [cryptoItemWithPrice, setCryptoItemWithPrice] =
+    useState<CryptoItemWithPrice | null>(null);
+
+  // EFFECTS
+  useEffect(() => {
+    // toggle modal visibility and background blur behind the modal
+    if (cryptoItemWithPrice !== null) {
+      setModalVisibility(true);
+      wrapperPrimaryRef.current?.classList.add("blur");
+    } else {
+      setModalVisibility(false);
+      wrapperPrimaryRef.current?.classList.remove("blur");
+    }
+  }, [cryptoItemWithPrice]);
+
+  // EVENT HANDLERS
+  const handleAddToPinned = async () => {
+    const newCryptoListToSuggest = cryptoListToSuggest.filter(
+      (oldCryptoItem: CryptoItem) => oldCryptoItem.id !== cryptoItem.id
+    );
+    const newCryptoListPinned = [...cryptoListPinned];
+    const result: any = await axios.get(
+      `${import.meta.env.VITE_API_CRYPTO_PRICE}${cryptoItem.id}-usd/spot`
+    );
+    newCryptoListPinned.push({
+      ...cryptoItem,
+      price: result.data.data.amount,
+    });
+    setCryptoListToSuggest(newCryptoListToSuggest);
+    setCryptoListPinned(newCryptoListPinned);
+    window.localStorage.setItem(
+      "cryptoListPinned",
+      JSON.stringify(newCryptoListPinned)
+    );
+  };
+
+  const handleViewPrice = () => {
+    wrapperPrimaryRef.current?.classList.add("blur");
+    axios
+      .get(`${import.meta.env.VITE_API_CRYPTO_PRICE}${cryptoItem.id}-usd/spot`)
+      .then((response) => {
+        setCryptoItemWithPrice({
+          ...cryptoItem,
+          price: response.data.data.amount,
+        });
+      })
+      .catch((error) => {
+        console.log(error);
```

Comment:
It is better to add a message also with the error when ever possible for better user experience. 
 - console.error("Error fetching crypto price:", error);